WebApplication/Startup.cs

```diff
@@ -105,8 +105,9 @@ public void ConfigureServices(IServiceCollection services)
             services.AddScoped<UserResolver>(); // TODO: use interface
             services.AddSingleton<LocalCache>();
             services.AddSingleton<Uploads>();
+            services.AddSingleton<AdoptProjectService>();
```

Comment:
It's NBD, but seems the business logic for POC can live in the controller, without new service. (I could be missing something though)

Reply:
I would like to have it this way. It splits Controller and Service. Enables me to write integration test for the adoption service and easily try the adoption. Moreover, it enables to split the work. 
I think it is worth very little (or none) overhead.